Default assertion message

It would be nice to provide some default messages when no mssage is provided, instead of showing:
```
java.lang.AssertionError: null
	at org.testng.Assert.fail(Assert.java:96)
	at org.testng.Assert.assertNotEquals(Assert.java:1157)
	at org.testng.Assert.assertNotEquals(Assert.java:1162)
```

For this example the Object::toString method could be used to output "%s not equals %s", ex:
```
    public static void assertNotEquals(Object actual1, Object actual2) {
        String message = String.format( "%s not equals %s", actual1, actual2 );
        assertNotEquals(actual1, actual2, message);
    }
```
Similar to other methods where no message is provided.

TestNG 6.14.3

--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -1,4 +1,5 @@
 Current
+New:   GITHUB-2118: Default assertion message (Jiong Fu)
 Fixed: GITHUB-2080: Wrong text for assertTrue (Jiong Fu)
 Fixed: GITHUB-2137: Issue with Priority with too low values (Krishnan Mahadevan)
 Fixed: GITHUB-2138: Factory calls are not intercepted by IAnnotationTransformer (Krishnan Mahadevan)

--- a/src/main/java/org/testng/Assert.java
+++ b/src/main/java/org/testng/Assert.java
@@ -1470,7 +1470,7 @@ public class Assert {
     }
 
     if (fail) {
-      Assert.fail(message);
+      Assert.fail(format(actual1, actual2, message, false));
     }
   }
 
@@ -1488,7 +1488,7 @@ public class Assert {
     }
 
     if (fail) {
-      Assert.fail(message);
+      Assert.fail(format(actual1, actual2, message, false));
     }
   }
 
@@ -1506,7 +1506,7 @@ public class Assert {
     }
 
     if (fail) {
-      Assert.fail(message);
+      Assert.fail(format(actual, expected, message, false));
     }
   }
 
@@ -1524,7 +1524,7 @@ public class Assert {
     }
 
     if (fail) {
-      Assert.fail(message);
+      Assert.fail(format(actual, expected, message, false));
     }
   }
 
@@ -1542,7 +1542,7 @@ public class Assert {
     }
 
     if (fail) {
-      Assert.fail(message);
+      Assert.fail(format(actual, expected, message, false));
     }
   }
 
@@ -1560,7 +1560,7 @@ public class Assert {
     }
 
     if (fail) {
-      Assert.fail(message);
+      Assert.fail(format(actual, expected, message, false));
     }
   }
 

--- a/src/test/java/org/testng/AssertTest.java
+++ b/src/test/java/org/testng/AssertTest.java
@@ -277,6 +277,50 @@ public class AssertTest {
     Assert.assertNotEquals("x", "x");
   }
 
+  @Test(description = "GITHUB-2118", expectedExceptions = AssertionError.class,
+      expectedExceptionsMessageRegExp = "did not expect \\[1.5\\] but found \\[1.5\\]")
+  public void testAssertNotEqualsFloat() {
+    Assert.assertNotEquals(1.5f, 1.5f, 0.5f);
+  }
+
+  @Test(description = "GITHUB-2118", expectedExceptions = AssertionError.class,
+      expectedExceptionsMessageRegExp = "did not expect \\[1.5\\] but found \\[1.5\\]")
+  public void testAssertNotEqualsDouble() {
+    Assert.assertNotEquals(1.5, 1.5, 0.5);
+  }
+
+  @Test(description = "GITHUB-2118", expectedExceptions = AssertionError.class,
+      expectedExceptionsMessageRegExp = "did not expect \\[\\[test\\]\\] but found \\[\\[test\\]\\]")
+  public void testAssertNotEqualsSet() {
+    Set<String> set = new HashSet<>();
+    set.add("test");
+    Assert.assertNotEquals(set, set);
+  }
+
+  @Test(description = "GITHUB-2118", expectedExceptions = AssertionError.class,
+      expectedExceptionsMessageRegExp = "did not expect \\[\\[test\\]\\] but found \\[\\[test\\]\\]")
+  public void testAssertNotEqualsDeepSet() {
+    Set<String> set = new HashSet<>();
+    set.add("test");
+    Assert.assertNotEqualsDeep(set, set);
+  }
+
+  @Test(description = "GITHUB-2118", expectedExceptions = AssertionError.class,
+      expectedExceptionsMessageRegExp = "did not expect \\[\\{key=value\\}\\] but found \\[\\{key=value\\}\\]")
+  public void testAssertNotEqualsMap() {
+    Map<String, String> map = new HashMap<>();
+    map.put("key", "value");
+    Assert.assertNotEquals(map, map);
+  }
+
+  @Test(description = "GITHUB-2118", expectedExceptions = AssertionError.class,
+      expectedExceptionsMessageRegExp = "did not expect \\[\\{key=value\\}\\] but found \\[\\{key=value\\}\\]")
+  public void testAssertNotEqualsDeepMap() {
+    Map<String, String> map = new HashMap<>();
+    map.put("key", "value");
+    Assert.assertNotEqualsDeep(map, map);
+  }
+
   class Contrived {
 
     int integer;

